src/menu.cpp

```diff
@@ -0,0 +1,120 @@
+#include "menu.h"
+
+const char* sceneToString(sceneNS::TYPE s) {
+	switch (s) {
+	case sceneNS::CLASSIC:			return "CLASSIC";
+	case sceneNS::TIME_ATK:			return "TIME ATTACK";
+	case sceneNS::HIGH_SCORES:		return "HIGH SCORES";
+	case sceneNS::CREDITS:			return "CREDITS";
+	}
+}
+
+Menu::Menu() {};
+
+Menu::Menu(Input* i, FontManager* fm) {
+	input = i;
+	baseFm = fm;
+}
+
+Menu::~Menu() {}
+
+void Menu::initialize() {
+	selectedItemIndex = 0;
+	blink = true;
+
+	// Copy from initialized fm
+	titleFm = new FontManager(*baseFm);
+	menuFm = new FontManager(*baseFm);
+
+	items.push_back(sceneNS::CLASSIC);
+	items.push_back(sceneNS::TIME_ATK);
+	items.push_back(sceneNS::HIGH_SCORES);
+	items.push_back(sceneNS::CREDITS);
+}
+
+void Menu::update(float frameTime) {
+	if (input->wasKeyPressed(UP_KEY)) {
+		selectedItemIndex = selectedItemIndex == 0 ? items.size() - 1 : selectedItemIndex - 1;
+	}
+
+	if (input->wasKeyPressed(DOWN_KEY)) {
+		selectedItemIndex = selectedItemIndex == items.size() - 1 ? 0 : selectedItemIndex + 1;
+	}
+
+	if (input->wasKeyPressed(ENTER_KEY)) {
+		nextSceneType = new sceneNS::TYPE;
+		nextSceneType = &items[selectedItemIndex];
```

Comment:
`nextSceneType` is initialized in the interface. Yeap, line 45 is needed, have to declare as a pointer of `sceneNS::Type` before assigning value.